src/ipc/ipc4/helper.c

```diff
@@ -160,6 +160,14 @@ static int ipc_pipeline_module_free(uint32_t pipeline_id)
 
 	icd = ipc_get_comp_by_ppl_id(ipc, COMP_TYPE_COMPONENT, pipeline_id);
 	while (icd) {
+		struct comp_buffer *sink;
+		struct list_item *sink_list;
+
+		list_for_item(sink_list, &icd->cd->bsink_list) {
```

Comment:
can we have a comment here. e.g. free all buffers because ...

Reply:
done you need to use the safe version here to iterate the list and delete the comp from the list as well? I'd assume you need to do the same even for iterating through the &ipc->comp_list as well as right?

Reply:
the sink_list will be release in the following component free. We don't need to do it in ipc->comp_list since buffer is not added into ipc->comp_list, as you know there is not buffer concept for ipc4

Reply:
In the future, we need to refine buffer usage to share one buffer for LL task

Reply:
This should really be a pipeline level API call
```c
pipeline_free_connected_buffers()
```
which in turn would be called by 
```c
pipeline_free()
```

Reply:
@lgirdwood  pipeline_free() is used by both IPC3 & IPC4, but pipeline_free_connected_buffers is only applied to IPC4

Reply:
two components in different pipeline will be ubind before deleting pipeline, so there is no such case "when two cores access the pipeline bsource_list/bsink_list at the same time on two different cores."